olive/passes/openvino/encapsulation.py

```diff
@@ -0,0 +1,226 @@
+# -------------------------------------------------------------------------
+# Copyright (c) Intel Corporation. All rights reserved.
+# Licensed under the MIT License.
+# --------------------------------------------------------------------------
+from pathlib import Path
+from typing import Dict, Type, Union
+from olive.hardware.accelerator import AcceleratorSpec, Device
+from olive.model import ONNXModelHandler, OpenVINOModelHandler
+from olive.passes import Pass
+from olive.passes.pass_config import BasePassConfig, PassConfigParam
+
+from openvino import get_version
+import onnx.helper as helper
+from onnx import TensorProto, save
+import shutil
+
+class OpenVINOEncapsulation(Pass):
+    openvino_to_onnx_dtype = {
+        'f32': TensorProto.FLOAT,
+        'float32': TensorProto.FLOAT,
+        'f64': TensorProto.DOUBLE,
+        'float64': TensorProto.DOUBLE,
+        'f16': TensorProto.FLOAT16,
+        'bf16': TensorProto.BFLOAT16,
+        'i8': TensorProto.INT8,
+        'int8_t': TensorProto.INT8,
+        'i16': TensorProto.INT16,
+        'int16_t': TensorProto.INT16,
+        'i32': TensorProto.INT32,
+        'int32_t': TensorProto.INT32,
+        'i64': TensorProto.INT64,
+        'int64_t': TensorProto.INT64,
+        'u8': TensorProto.UINT8,
+        'uint8_t': TensorProto.UINT8,
+        'u16': TensorProto.UINT16,
+        'uint16_t': TensorProto.UINT16,
+        'u32': TensorProto.UINT32,
+        'uint32_t': TensorProto.UINT32,
+        'u64': TensorProto.UINT64,
+        'uint64_t': TensorProto.UINT64,
+        'bool': TensorProto.BOOL,
+        'boolean': TensorProto.BOOL,
+        # Add more if needed
+    }
+    # Add any required data members to the class
+    @classmethod
+    def _default_config(cls, accelerator_spec: AcceleratorSpec) -> Dict[str, PassConfigParam]:
+        return {
+            "target_device": PassConfigParam(
+                type_=Device,
+                default_value=accelerator_spec.accelerator_type.CPU,
+                required=False,
+                description=(
+                    "Device the encapsulated model should run on."
+                    "Available devices are cpu, gpu, npu."
+                ),
+            ),
+            "shared_cache": PassConfigParam(
```

Comment:
This term is used by Olive cache system already. For better understanding, can you use other names for this? Like `reuse_cache`?